### PR TITLE
Add spill metric spillExtractVectorTimeNanos

### PR DESCRIPTION
**Pull request by jinchengchenghh**
The extract vector time is 315.24ms while sort time is 475.01ms with default row type `INTEGER(), BIGINT(), VARCHAR(), VARBINARY(), DOUBLE()` and key type `INTEGER(), BIGINT()`

### CODE REVIEW

**Comment by jinchengchenghh** [comment]
/mnt/DP_disk1/code/velox/build/velox/exec/tests# ./velox_spiller_aggregate_benchmark -spiller_benchmark_spill_executor_size=1
WARNING: Logging before InitGoogleLogging() is written to STDERR
I0920 13:26:08.279875 3168791 AggregateSpillBenchmarkBase.cpp:81] ======Aggregate AGGREGATE_INPUT spilling statistics======
I0920 13:26:08.280012 3168791 AggregateSpillBenchmarkBase.cpp:83] total execution time: 714.70ms
I0920 13:26:08.280053 3168791 AggregateSpillBenchmarkBase.cpp:84] 10000 vectors each with 100 rows have been processed
I0920 13:26:08.280063 3168791 AggregateSpillBenchmarkBase.cpp:87] peak memory usage[0B] cumulative memory usage[0B]
I0920 13:26:08.280081 3168791 AggregateSpillBenchmarkBase.cpp:90] spillRuns[1] spilledInputBytes[1.49GB] spilledBytes[122.09MB] spilledRows[1000000] spilledPartitions[8] spilledFiles[8] spillFillTimeNanos[27.04ms] spillSortTimeNanos[475.01ms] spillExtractVectorTime[315.24ms] spillSerializationTimeNanos[46.13ms] spillWrites[128] spillFlushTimeNanos[11.79ms] spillWriteTimeNanos[76.20ms] maxSpillExceededLimitCount[0] spillReadBytes[0B] spillReads[0] spillReadTimeNanos[0ns] spillReadDeserializationTimeNanos[0ns]
I0920 13:26:08.280377 3168791 AggregateSpillBenchmarkBase.cpp:97] spilled file /tmp/velox_test_YxjXUk/SpillerBenchmarkTest-spill-7-0-7 size 15.12MB
I0920 13:26:08.280404 3168791 AggregateSpillBenchmarkBase.cpp:97] spilled file /tmp/velox_test_YxjXUk/SpillerBenchmarkTest-spill-0-0-0 size 15.30MB
I0920 13:26:08.280423 3168791 AggregateSpillBenchmarkBase.cpp:97] spilled file /tmp/velox_test_YxjXUk/SpillerBenchmarkTest-spill-2-0-2 size 15.28MB
I0920 13:26:08.280443 3168791 AggregateSpillBenchmarkBase.cpp:97] spilled file /tmp/velox_test_YxjXUk/SpillerBenchmarkTest-spill-4-0-4 size 15.37MB
I0920 13:26:08.280462 3168791 AggregateSpillBenchmarkBase.cpp:97] spilled file /tmp/velox_test_YxjXUk/SpillerBenchmarkTest-spill-3-0-3 size 15.23MB
I0920 13:26:08.280481 3168791 AggregateSpillBenchmarkBase.cpp:97] spilled file /tmp/velox_test_YxjXUk/SpillerBenchmarkTest-spill-1-0-1 size 15.35MB
I0920 13:26:08.280500 3168791 AggregateSpillBenchmarkBase.cpp:97] spilled file /tmp/velox_test_YxjXUk/SpillerBenchmarkTest-spill-5-0-5 size 15.22MB
I0920 13:26:08.280519 3168791 AggregateSpillBenchmarkBase.cpp:97] spilled file /tmp/velox_test_YxjXUk/SpillerBenchmarkTest-spill-6-0-6 size 15.22MB
I0920 13:26:08.280552 3168791 SpillerBenchmarkBase.cpp:143] Remove spill dir: /tmp/velox_test_YxjXUk
I0920 13:26:08.304075 3168791 TempDirectoryPath.cpp:29] TempDirectoryPath:: removing all files from /tmp/velox_test_YxjXUk

**Review by xiaoxmeng — COMMENTED** [review]
@jinchengchenghh thanks for the change. So is there any benefit to pipeline vector extraction + serialization/remote disk io? Thanks!

**Comment by jinchengchenghh** [comment]
The default value to extract rows from RowContainer is 64, with this metric, we can measure the extract strategy and extract potential optimization. It is really a high expense in spill.

**Comment by jinchengchenghh** [comment]
Ideally, if we can split the total spill process to some parts, and report the time of each part, it may help optimize spill.

**Comment by jinchengchenghh** [comment]
```
WARNING: Logging before InitGoogleLogging() is written to STDERR
I0924 09:27:18.094496 3456531 AggregateSpillBenchmarkBase.cpp:81] ======Aggregate AGGREGATE_INPUT spilling statistics======
I0924 09:27:18.094645 3456531 AggregateSpillBenchmarkBase.cpp:83] total execution time: 720.74ms
I0924 09:27:18.094692 3456531 AggregateSpillBenchmarkBase.cpp:84] 10000 vectors each with 100 rows have been processed
I0924 09:27:18.094702 3456531 AggregateSpillBenchmarkBase.cpp:87] peak memory usage[0B] cumulative memory usage[0B]
I0924 09:27:18.094718 3456531 AggregateSpillBenchmarkBase.cpp:90] spillRuns[1] spilledInputBytes[1.49GB] spilledBytes[122.09MB] spilledRows[1000000] spilledPartitions[8] spilledFiles[8] spillFillTimeNanos[23.76ms] spillSortTimeNanos[474.75ms] spillExtractVectorTime[168.37ms] spillSerializationTimeNanos[47.05ms] spillWrites[128] spillFlushTimeNanos[12.90ms] spillWriteTimeNanos[79.89ms] maxSpillExceededLimitCount[0] spillReadBytes[0B] spillReads[0] spillReadTimeNanos[0ns] spillReadDeserializationTimeNanos[0ns]
I0924 09:27:18.094986 3456531 AggregateSpillBenchmarkBase.cpp:97] spilled file /tmp/velox_test_UswECP/SpillerBenchmarkTest-spill-7-0-7 size 15.12MB
I0924 09:27:18.095016 3456531 AggregateSpillBenchmarkBase.cpp:97] spilled file /tmp/velox_test_UswECP/SpillerBenchmarkTest-spill-0-0-0 size 15.30MB
I0924 09:27:18.095034 3456531 AggregateSpillBenchmarkBase.cpp:97] spilled file /tmp/velox_test_UswECP/SpillerBenchmarkTest-spill-2-0-2 size 15.28MB
I0924 09:27:18.095053 3456531 AggregateSpillBenchmarkBase.cpp:97] spilled file /tmp/velox_test_UswECP/SpillerBenchmarkTest-spill-4-0-4 size 15.37MB
I0924 09:27:18.095072 3456531 AggregateSpillBenchmarkBase.cpp:97] spilled file /tmp/velox_test_UswECP/SpillerBenchmarkTest-spill-3-0-3 size 15.23MB
I0924 09:27:18.095091 3456531 AggregateSpillBenchmarkBase.cpp:97] spilled file /tmp/velox_test_UswECP/SpillerBenchmarkTest-spill-1-0-1 size 15.35MB
I0924 09:27:18.095110 3456531 AggregateSpillBenchmarkBase.cpp:97] spilled file /tmp/velox_test_UswECP/SpillerBenchmarkTest-spill-5-0-5 size 15.22MB
I0924 09:27:18.095130 3456531 AggregateSpillBenchmarkBase.cpp:97] spilled file /tmp/velox_test_UswECP/SpillerBenchmarkTest-spill-6-0-6 size 15.22MB
I0924 09:27:18.095162 3456531 SpillerBenchmarkBase.cpp:143] Remove spill dir: /tmp/velox_test_UswECP
I0924 09:27:18.120522 3456531 TempDirectoryPath.cpp:29] TempDirectoryPath:: removing all files from /tmp/velox_test_UswECP
```

File: velox/exec/Spiller.cpp [hunk 1]
```diff
@@ -306,22 +306,26 @@ int64_t Spiller::extractSpillVector(
     RowVectorPtr& spillVector,
     size_t& nextBatchIndex) {
   VELOX_CHECK_NE(type_, Type::kHashJoinProbe);
-
+  uint64_t extractNanos{0};
```
Comment (xiaoxmeng) [inline velox/exec/Spiller.cpp:309]:
```
nit: s/extractNanos/extractNs/
```

**Review by xiaoxmeng — APPROVED** [review]
@jinchengchenghh LGTM. Thanks!

**Comment by xiaoxmeng** [comment]
> The default value to extract rows from RowContainer is 64, with this metric, we can measure the extract strategy and extract potential optimization. It is really a high expense in spill.

@jinchengchenghh let's check the cost of meta internal workload cc @tanjialiang @duanmeng

**Pull request closed without merging**